packages/webapp/src/components/Filter/FilterMultiSelectV2/index.tsx

```diff
@@ -0,0 +1,251 @@
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { MouseEvent, useEffect, useLayoutEffect, useMemo, useRef } from 'react';
+import Select, {
+  ClearIndicatorProps,
+  components,
+  MenuListProps,
+  MultiValue,
+  MultiValueRemoveProps,
+  OptionProps,
+} from 'react-select';
+import { ComponentFilter, ComponentFilterOption } from '../types';
+import { useState } from 'react';
+import { ReactComponent as XCircle } from '../../../assets/images/x-circle.svg';
+import { useTranslation } from 'react-i18next';
+import TextButton from '../../Form/Button/TextButton';
+import styles from './styles.module.scss';
+import Checkbox from '../../Form/Checkbox';
+import produce from 'immer';
+import { FilterItemProps } from '../FilterGroup';
+import { Label } from '../../Typography';
+
+interface FilterMultiSelectProps extends ComponentFilter {
+  onChange?: FilterItemProps['onChange'];
+  isDisabled?: boolean;
+}
+
+const MultiValueRemove = (props: MultiValueRemoveProps) => (
+  <components.MultiValueRemove {...props}>
+    <XCircle />
+  </components.MultiValueRemove>
+);
+
+const ClearIndicator = (props: ClearIndicatorProps<ComponentFilterOption>) => {
+  const { getValue, clearValue } = props;
+  const value = getValue();
+  const { t } = useTranslation();
+
+  const onClick = (e: MouseEvent) => {
+    e.stopPropagation();
+    e.preventDefault();
+    clearValue();
+  };
+
+  return (
+    <>
+      {t('FILTER.SELECTED', { count: value.length })} -
+      <TextButton className={styles.clearTextButton} onClick={onClick}>
+        clear
+      </TextButton>
+    </>
+  );
+};
+
+const Option = (props: OptionProps<ComponentFilterOption>) => {
+  const { label, isSelected, selectOption, data } = props;
+
+  const onClick = (e: MouseEvent) => {
+    selectOption(data);
+    e.stopPropagation();
+    e.preventDefault();
+  };
+
+  return (
+    <components.Option {...props}>
+      <TextButton className={styles.optionButton} onClick={onClick}>
+        <Checkbox label={label} checked={isSelected} />
+      </TextButton>
+    </components.Option>
+  );
+};
+
+const MenuList = (props: MenuListProps<ComponentFilterOption>) => {
+  const { children, getValue, hasValue, options, clearValue } = props;
+  const { t } = useTranslation();
+
+  const partiallyChecked = getValue().length !== options.length;
+
+  return (
+    <components.MenuList {...props}>
+      <div className={styles.option}>
+        <TextButton className={styles.optionButton} onClick={clearValue}>
```

Comment:
And I _think_ -- this is what I understand from [this conversation]( https://www.figma.com/file/eMffC0h3JPlWtzNttxJpil/Litefarm-Updated?type=design&node-id=2805-37627&mode=design&t=mloVzFJhA0P2XRXV-0) between Loïc and Sayaka -- that it can just be plain non-interactable; doesn't need an onClick handler and can probably get `pointer-events: none;'

Reply:
Actually let me also tag @Duncan-Brain on that since it was relevant to what he said above (and the interaction he shows in the video where clicking the "All selected" row does nothing but close the dropdown was also the first thing I noticed and was bothered by 😉).

I think full spec according to the linked Figma is no interaction on that indicator row (clearing or otherwise) and should open with default state checked and selected.

Reply:
I thought we had mentioned keeping it interaction free at some point too but based on this in Figma

<img width="680" alt="Captura de pantalla 2024-03-27 a la(s) 21 41 58" src="https://github.com/LiteFarmOrg/LiteFarm/assets/20230771/6aa32bfb-7cc7-4d0c-a232-a5503e42a64a">

I thought it does need to clear the filters if not in the default state? Might be that's just outdated though

Reply:
I've gone ahead and made this non-interactive for now to avoid confusion, should be easy to adjust down the road if needed!